### DISCUSSION TITLE
Homer7 and Kamailio 5.6 setup issues

### DISCUSSION THREAD

**Discussion by hopefulGiupplo** [sipcapture/homer#619] [Q&A] [answered]
Hello, I'm having some issues with Homer7. My Kamailio 5.6.0 vm is **correctly sending hep3** packets to the heplify-server in the homer environment, which are correctly received - I checked with tcpdump and wireshark. However, I don't understand why **the dbs are still empty**.
The same setup is working if I'm using heplify to send hep packets, but I'm having trouble finding documentation that explains if hep packets from kamailio need to be processed in a different way, or if Homer needs some options to be turned on/off.
Thank you in advance for your help!

**Comment by lmangani** [comment]
HEP packets are all the same, there's no difference in how they are handled. 
Just make sure you use the correct protocol. We suggest using TCP/HEP whenever possible. Also make sure there are no firewall or selinux rules dropping the packets - arriving packets would show on tcpdumps but still be prevented from reaching the local socket.

**Reply by hopefulGiupplo** [reply]
Thank you for your rapid response, I'll check if it's a firewall issue. In that case, could heplify-server logs show incoming packets anyway?

**Reply by lmangani** [reply]
If the packets are dropped by the firewall/system heplify-server would not see them - only tcpdump/ngrep would at the ingress if they are received and dropped by selinux or what not. If heplify-sees them, the issue is elsewhere and probably could be related to the timestamp, etc.

**Reply by hopefulGiupplo** [reply]
Hello, sorry fot not responding right away, I did some tests to be sure I was doing things right. Me and my collegues are pretty sure that there are no firewall issues, but we've managed to understand some additional information.

- Kamailio sends HEP3/SIP packet to the right socket of heplify-server, and those packets are recognized by wireshark. However, sometimes heplify-server logs a WARN saying it receives malformed packets. I'll include my kamailio setup below, but let me know if this should be an issue for the Kamailio mailing list, rather than here.
- A tcpdump on the homer machine shows that packets arrive, and wireshark can decode the pcap capture without problems
- the Heplify-server doesn't even try to insert the packets it receives into the postgres database, seems like it ditches them.
- Using heplify on kamailio machine, instead of siptrace, everything works.

My kamailio setup:
```
# ----- siptrace params -----
#!ifdef WITH_HOMER
modparam("siptrace", "duplicate_uri", "sip:MY_HOMER_MACHINE:9060")
modparam("siptrace", "send_sock_address", "sip:MY_KAMAILIO_MACHINE:5060")
modparam("siptrace", "hep_mode_on", 1)
modparam("siptrace", "hep_version", 3)
modparam("siptrace", "trace_to_database", 0)
modparam("siptrace", "trace_mode", 5)
modparam("siptrace", "trace_on", 1)
#!endif
```
Routing logic:

```
request_route {
    #!ifdef WITH_HOMER
    sip_trace_mode("t");
    hlog("Logging test.");
    #!endif
...
}

```

**Comment by lmangani** [comment]
This is now an issue rather than a discussion. Please open an issue with all the details

> heplify-server logs a WARN saying it receives malformed packets

If packets are malformed we need samples to replicate the issue. we cannot guess what's wrong with them otherwise.

> A tcpdump on the homer machine shows that packets arrive, and wireshark can decode the pcap capture without problems

Please provide some export in your issue from the raw pcap for the failed inserts causing the warning

**Reply by hopefulGiupplo** [reply]
Thank you, I'll open the issue.

**Comment by ostap0207** [comment]
> We suggest using TCP/HEP whenever possible

Hello, what configuration options specifies to use TCP/HEP instead of UDP?

**Reply by adubovikov** [reply, marked as answer]
Just add transport=tcp as an uri param (;transport=tcp)

Here is an example:
`modparam("siptrace", "duplicate_uri", "sip:MY_HOMER_MACHINE:9060;transport=tcp")`
